WT-11713 Check min_split_size when growing the split buffer in __wt_rec_split

y123456yz opened the pull request:
__wt_rec_split buf fix, we shoud use min_split_size, not use split_size / 2, it is more accurate by using min_split_size

ajmorton commented:
Thanks for your submission @y123456yz! The proposed change looks good, we're just hitting a few test failures as `r->min_split_size` isn't used for Fixed Length Column Store. It's [expected to be zero](https://github.com/wiredtiger/wiredtiger/blob/b6b78206e16bf421c998c5c8656896df6e6fa9f4/src/reconcile/rec_write.c#L1534) however we haven't [initialized it to zero in __wt_rec_split_init](https://github.com/wiredtiger/wiredtiger/blob/b6b78206e16bf421c998c5c8656896df6e6fa9f4/src/reconcile/rec_write.c#L1139C1-L1139C1). 

Would you mind adding lines [here](https://github.com/wiredtiger/wiredtiger/blob/b6b78206e16bf421c998c5c8656896df6e6fa9f4/src/reconcile/rec_write.c#L1139C1-L1139C1) that set `min_split_size` and `min_space_avail` to zero with a comment explaining the fields are always zero for FLCS? Alternatively if you'd like to provide [edit permissions on your branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) I can make these changes for you.

y123456yz commented:
@ajmorton 

![image](https://github.com/wiredtiger/wiredtiger/assets/14356223/4da085da-985e-42f9-a624-e431af985428)

According to your suggestion, I have modified the relevant code and added comments.

If there is any unreasonable place, you can directly edit and modify.

thanks .

ajmorton reviewed (approved):
The change looks good and the contributors agreement has been signed.
Thanks for submitting!

ajmorton commented:
evergreen merge

ajmorton commented:
@y123456yz we've reverted this commit due to reliable failures in our `test-prepare-hs03-hook-timestamp`. This is being investigated in WT-11741

y123456yz commented:
> @y123456yz we've reverted this commit due to reliable failures in our `test-prepare-hs03-hook-timestamp`. This is being investigated in [WT-11741](https://jira.mongodb.org/browse/WT-11741)

got it, thanks, I keep an eye on it too.
